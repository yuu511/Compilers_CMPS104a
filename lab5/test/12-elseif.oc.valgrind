==5832== Memcheck, a memory error detector
==5832== Copyright (C) 2002-2017, and GNU GPL'd, by Julian Seward et al.
==5832== Using Valgrind-3.13.0 and LibVEX; rerun with -h for copyright info
==5832== Command: ../oc 12-elseif.oc
==5832== 
==5832== 
==5832== HEAP SUMMARY:
==5832==     in use at exit: 0 bytes in 0 blocks
==5832==   total heap usage: 472 allocs, 472 frees, 145,020 bytes allocated
==5832== 
==5832== All heap blocks were freed -- no leaks are possible
==5832== 
==5832== For counts of detected and suppressed errors, rerun with: -v
==5832== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
