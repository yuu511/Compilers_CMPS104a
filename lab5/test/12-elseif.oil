main:      .function int 
           .param int argc
.if0:      goto .el0 if argc != 1
.th0:      call putstr("one")  
.fi0       goto   
.el0:
.if1:      goto .el1 if argc != 2
.th1:      call putstr("two")  
.fi1       goto   
.el1:
.if2:      goto .el2 if argc != 3
.th2:      call putstr("three")  
.fi2       goto   
.el2:      call putstr("many")  
.fi2:
.fi1:
.fi0:
           call putchr('\n')  
           return 0
           return
           .end
