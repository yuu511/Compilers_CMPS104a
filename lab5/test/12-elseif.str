string_set[   0]:    7296690267450279874 0x558820fa1cd8->"chr"
                   13348923631108115385 0x558820fa2698->"putstr"
string_set[   5]:   12679825168387489523 0x558820fa21f8->"num"
string_set[   6]:    9317282556992397945 0x558820fa38d8->"if"
string_set[   7]:    2297668033614959926 0x558820fa5078->"0"
string_set[  10]:    1583576014053361350 0x558820fa2858->"str"
                   13497578857207247412 0x558820fa1278->"string"
string_set[  13]:   12868445110721718657 0x558820fa18f8->")"
string_set[  16]:   11911796782241914524 0x558820f9ffe8->"void"
                   10159970873491820195 0x558820fa3b58->"1"
                   11780556996745140861 0x558820fa4998->""three""
string_set[  19]:    5344662657899890615 0x558820fa19b8->";"
string_set[  20]:   15816506932648033945 0x558820fa11b8->"("
                   11868120863286193964 0x558820fa3478->"main"
string_set[  25]:   11166701219533395393 0x558820fa2e78->"getstr"
string_set[  29]:   14635384904416252381 0x558820fa4db8->"return"
string_set[  32]:   13084547564353593782 0x558820fa43d8->""two""
string_set[  34]:    7211130878751977441 0x558820fa1b18->"putchr"
string_set[  35]:   14736352883882276958 0x558820fa3a98->"=="
string_set[  37]:   12291392672316406109 0x558820fa1818->"line"
                    6192799629375248351 0x558820fa4d58->"'\n'"
                   15712226061040878727 0x558820fa3dd8->""one""
string_set[  39]:   18413295556186398326 0x558820fa2038->"putint"
string_set[  44]:      21606623943142546 0x558820fa3818->"{"
string_set[  45]:    6142509188972423790 0x558820fa13f8->""
string_set[  47]:   13686721134187162920 0x558820fa50e8->"}"
string_set[  51]:    3701972582333163920 0x558820fa3638->"argc"
                    9509744370891421448 0x558820fa4bf8->""many""
string_set[  57]:    9386170080689500517 0x558820fa1338->"expr"
string_set[  58]:    4017852018407011111 0x558820fa15d8->"file"
                    3729804957429652673 0x558820f9acc8->"<<ROOT>>"
string_set[  59]:   16579916199511856337 0x558820fa3178->"getln"
string_set[  61]:    4551451650890805270 0x558820fa4178->"2"
string_set[  63]:    9611428142580219387 0x558820fa10b8->"fail"
string_set[  64]:   12597418120160561823 0x558820fa1758->"int"
string_set[  69]:    1873634190020599143 0x558820fa3eb8->"else"
string_set[  70]:    9201800968376702500 0x558820fa2b78->"getchr"
string_set[  74]:    8248777770799913213 0x558820fa4738->"3"
string_set[  78]:    4706206694200543916 0x558820fa10f8->","
load_factor = 0.494
bucket_count = 79
max_bucket_size = 3
