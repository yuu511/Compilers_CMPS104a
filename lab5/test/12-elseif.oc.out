CONDITIONAL IF STATEMENT SYMBOL TOK_EQ 
CONDITIONAL IF STATEMENT SYMBOL TOK_EQ 
CONDITIONAL IF STATEMENT SYMBOL TOK_EQ 
